src/screens/Profile/TravelToken/index.tsx

```diff
@@ -1,22 +1,46 @@
-import {Edit} from '@atb/assets/svg/mono-icons/actions';
 import FullScreenHeader from '@atb/components/screen-header/full-header';
-import * as Sections from '@atb/components/sections';
-import ThemeText from '@atb/components/text';
-import ThemeIcon from '@atb/components/theme-icon/theme-icon';
-import {useMobileTokenContextState} from '@atb/mobile-token/MobileTokenContext';
 import {StyleSheet, Theme} from '@atb/theme';
 import {TravelTokenTexts, useTranslation} from '@atb/translations';
 import TravelTokenBox from '@atb/travel-token-box';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {View} from 'react-native';
 import {ScrollView} from 'react-native-gesture-handler';
 import {ProfileScreenProps} from '../types';
+import {FaqSection} from '@atb/screens/Profile/TravelToken/FaqSection';
+import {ChangeTokenAction} from '@atb/screens/Profile/TravelToken/ChangeTokenAction';
+import {useMobileTokenContextState} from '@atb/mobile-token/MobileTokenContext';
+import {useIsFocused} from '@react-navigation/native';
 
 type TravelCardScreenProps = ProfileScreenProps<'TravelToken'>;
 
 export default function TravelCard({navigation}: TravelCardScreenProps) {
   const styles = useStyles();
   const {t} = useTranslation();
+  const [showLoader, setShowLoader] = useState<boolean>(false);
+  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
+  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();
+
+  const {isError, isLoading, getTokenToggleDetails} =
+    useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
```

Comment:
Maybe more correct to use [`useFocusEffect`](https://reactnavigation.org/docs/use-focus-effect) instead of [`useIsFocused`](https://reactnavigation.org/docs/use-is-focused) here, since the `screenHasFocus` triggers a `useEffect`.

Reply:
```
  const shouldFetchTokenDetails = !isError && !isLoading;
  const [showLoader, setShowLoader] = useState<boolean>(false);
  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();

  useFocusEffect(useCallback(
      () => {
        const {showLoader, toggleLimit, maxToggleLimit} = useTokenToggleDetails(
            shouldFetchTokenDetails,
        );
        setShowLoader(showLoader)
        setMaxToggleLimit(maxToggleLimit)
        setToggleLimit(toggleLimit)
      },
      [shouldFetchTokenDetails],
  ))
```

I tried to change it. This will now not re-render the component on every focus, but will still do when shouldFetchTokenDetails changes which I believe should not change a lot. Although when we change to useFocusEffect, we have to maintain the state here as well along with the hook so the code becomes slightly verbose rather than what it is now.  How do you think it looks here ☝️  vs what it is now?

Reply:
No worries. Keep it at it is!